src/test/java/bio/terra/service/snapshotbuilder/query/QueryTest.java

```diff
@@ -25,4 +27,80 @@ public static Query createQuery() {
   void renderSQL() {
     assertThat(createQuery().renderSQL(null), is("SELECT t.* FROM table AS t"));
   }
+
+  @Test
```

Comment:
@okotsopoulos suggested we might not want this test actually in our unit test suite because it might be hard to debug. I think the `isDistinct` work is valuable, and this is useful documentation, but we may not want to actually merge this code. @pshapiro4broad, what do you think?